### PR TITLE
Omit direct write access to HttpResponseWriter

### PR DESCRIPTION
**Pull request by crissi98**
fixes sonar findings: https://lift.sonatype.com/result/HUK-COBURG/openapirouter/01F8F9RTWEDVSQYXP9GS4HCY6B?tab=results

### CODE REVIEW

File: response.go [hunk 1]
```diff
@@ -27,15 +28,15 @@ func (response *Response) write(writer http.ResponseWriter) {
 	case string:
 		writer.Header().Set("Content-Type", "text/plain; charset=utf-8")
 		writer.WriteHeader(response.StatusCode)
-		_, err = writer.Write([]byte(data))
+		_, err = io.WriteString(writer, data)
```
[the remark this answers is not in the document]
Comment (crissi98) [reply response.go:31]: @Lift-def ignore
a plain text response should be written so no html escaping is necessary

[the remark this answers is not in the document]
Comment (cniweb) [reply response.go:31]: @Lift-dev ignore

File: response.go [hunk 2]
```diff
@@ -27,15 +28,15 @@ func (response *Response) write(writer http.ResponseWriter) {
...
 	default:
 		if data != nil {
 			writer.Header().Set("Content-Type", "application/json; charset=utf-8")
 			writer.WriteHeader(response.StatusCode)
 			err = json.NewEncoder(writer).Encode(response.Body)
 		} else {
 			writer.WriteHeader(response.StatusCode)
-			_, err = writer.Write(make([]byte, 0))
+			_, err = io.WriteString(writer, "")
```
Comment (cniweb) [inline response.go:39]: @Lift-dev ignore

File: response.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package openapirouter
 
 import (
 	"encoding/json"
+	"io"
```
Comment (cniweb) [inline response.go:5]: @Lift-dev ignore

**Review by cniweb — CHANGES_REQUESTED** [review]

**Comment by svnhof** [comment]
@crissi98 Since we only work with JSON here I see no need for the change

**Comment by crissi98** [comment]
@svnhof You are right. And since this PR does not fix the issue, so I am closing it.

**Pull request closed without merging**